FrenetOptimalTrajectory/fot.py

```diff
@@ -89,33 +98,151 @@ def main():
-            plt.title("v[m/s]:" + str(
-                      np.linalg.norm(initial_conditions['vel']))[0:4]
-            )
+            plt.title("v[m/s]:" +
+                      str(np.linalg.norm(initial_conditions['vel']))[0:4])
             plt.grid(True)
-            plt.savefig("{}.jpg".format(i))
+            if save_frame:
+                Path("img/frames").mkdir(parents=True, exist_ok=True)
+                plt.savefig("img/frames/{}.jpg".format(i))
             plt.pause(0.1)
 
     print("Finish")
+
+    print("======================= SUMMARY ========================")
+    print("Total time for {} iterations taken: {}".format(i, total_time))
     print("Average time per iteration: {}".format(total_time / i))
+    print("Max time per iteration: {}".format(max(time_list)))
+
+    return time_list
+
 
 if __name__ == '__main__':
-    main()
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-d",
+        "--display",
+        action="store_true",
+        help="show animation, ensure you have X11 forwarding server open")
+    parser.add_argument("-v",
+                        "--verbose",
+                        action="store_true",
+                        help="verbose mode, show all state info")
+    parser.add_argument("-s",
+                        "--save",
+                        action="store_true",
+                        help="save each frame of simulation")
+    parser.add_argument(
+        "-c",
+        "--compare",
+        action="store_true",
+        help="compare threaded runtime with unthreaded baseline")
+    parser.add_argument("-t",
+                        "--thread",
+                        type=int,
+                        default=0,
+                        help="set number of threads to run with")
+    parser.add_argument("-p",
+                        "--profile",
+                        action="store_true",
+                        help="enable time profiling")
+    parser.add_argument("-f",
+                        "--full",
+                        action="store_true",
+                        help="full time profiling")
+    args = parser.parse_args()
+
+    # run planner with args passed in, record time
+    planner_time = run_fot(args.display, args.verbose, args.thread, args.save)
+
+    if args.compare:
+        baseline_time = run_fot(False, False, 0, False)
+
+        print("======================= SPEED UP ========================")
+        print("Average Speed Up per Iteration: {} x".format(
+            np.mean(baseline_time) / np.mean(planner_time)))
+
+    if args.profile:
+        if args.compare:
+            plt.plot(baseline_time, color="k", label="Baseline")
+            plt.hlines(np.mean(baseline_time),
+                       0,
+                       len(baseline_time),
+                       colors="k",
+                       label="Baseline mean",
+                       linestyles="dashed")
+
+        plt.plot(planner_time,
```

Comment:
Can you please move all the plotting code to a separate script that processes the logs? The fot.py should only contain code necessary to run fot, whereas the other script should contain code to run the experiments and plot the results.

Reply:
Good point, will do!